src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMBuilder.java

```diff
@@ -129,32 +127,9 @@ public static UriTemplate checkoutUriTemplate(@CheckForNull Item context,
                     )
             );
             if (credentials instanceof SSHUserPrivateKey) {
-                URI sshUri = URI.create(sshRemote);
-                return UriTemplate.buildFromTemplate(
-                        "ssh://git@" + sshUri.getHost() + (sshUri.getPort() != 22 ? ":" + sshUri.getPort() : ""))
-                        .path(UriTemplateBuilder.var("projectPath"))
-                        .literal(".git")
+                return UriTemplate.buildFromTemplate("ssh://" + sshRemote)
```

Comment:
not sure where sshRemote comes from but the project API returns both `ssh_url_to_repo` and `http_url_to_repo`

Reply:
Yes, that is true. 

`if(sshRemote != null)` is an additional check. If deemed unnecessary, can be removed.

If the checkout credential is of `SSHUserPrivateKey` type we use the `sshRemote` url else the http server url.

`sshRemote` is populated in https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/c370e20fe8412327ea21bbd0a50a2425fea453fd/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java#L214

Reply:
Actually `sshRemote` is checked null because, when we call `GitSCMBuilder` constructor inside `GitLabSCMBuilder`'s constructor we need the `http url`. So in the call to `checkoutUriTemplate` sshRemote is passed as `null`.